wgpu-hal/src/vulkan/device.rs

```diff
@@ -725,6 +725,15 @@ impl crate::Device<super::Api> for super::Device {
         &self,
         desc: &crate::BufferDescriptor,
     ) -> Result<super::Buffer, crate::DeviceError> {
+        if self
+            .shared
+            .workarounds
+            .contains(super::Workarounds::BUFFER_EXTENTS_I32)
+            && desc.size > std::i32::MAX as wgt::BufferAddress
+        {
+            return Err(crate::DeviceError::OutOfMemory);
```

Comment:
We should still reduce the maximum buffer size limit to 2GB so that we don't need this check here and the user is aware of the hard limit